### ISSUE TITLE
auto-detect module dependencies and do something with it

### ISSUE THREAD

**Issue by VanCoding** [VanCoding/task-graph-processor#6]

**Comment by VanCoding** [comment]
Realization:
There's a difference between package dependencies and task dependencies!

Task dependencies should only be tasks that **need** to run for the dependent taks to be successful. An example for this is the building. Dependencies need to be built in order to to be able to import them.

Then there are package dependencies. Maybe a good example for this are unit tests. A unit test should be able to run without running the unit tests of the dependencies as well. It of course it might make a lot of sense to also run all the tests of the dependencies, but it should not be strictly coupled.

The bottom line is:
Task dependencies should always run. For package dependencies, the task runner should  make it possible to run the task in the dependency-packages as well. These should be able to run in parallel.
